solid/src/main/java/com/inrupt/client/solid/SolidClient.java

```diff
@@ -280,14 +285,15 @@ public <T extends Resource> CompletionStage<Void> delete(final T resource, final
         defaultHeaders.firstValue(USER_AGENT).ifPresent(agent -> builder.setHeader(USER_AGENT, agent));
         headers.firstValue(USER_AGENT).ifPresent(agent -> builder.setHeader(USER_AGENT, agent));
 
-        return client.send(builder.build(), Response.BodyHandlers.ofByteArray()).thenApply(res -> {
-            if (isSuccess(res.statusCode())) {
-                return null;
-            } else {
-                throw SolidClientException.handle("Unable to delete resource", resource.getIdentifier(),
-                        res.statusCode(), res.headers(), new String(res.body(), UTF_8));
-            }
-        });
+        return client.send(
+            builder.build(),
+            Response.BodyHandlers.throwOnError(
+                Response.BodyHandlers.ofByteArray(),
+                (r) -> isSuccess(r.statusCode()),
+                httpExceptionMapper("Deleting resource " + resource.getIdentifier() + " failed.")
+            )
+        )// FIXME I don't understand why the following is required.
+        .thenAccept(o -> { /* no-op */ });
```

Comment:
I don't know that this is the correct approach

Reply:
It's because the method signature is (async) `void`.

Previously (when error handling was here) the response from the body handler was mapped (`thenApply`) to nothing.